exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/shared/ddl/ColumnTransformTest.kt

```diff
@@ -0,0 +1,131 @@
+package org.jetbrains.exposed.sql.tests.shared.ddl
+
+import org.jetbrains.exposed.dao.id.IntIdTable
+import org.jetbrains.exposed.sql.ColumnTransformer
+import org.jetbrains.exposed.sql.StdOutSqlLogger
+import org.jetbrains.exposed.sql.addLogger
+import org.jetbrains.exposed.sql.insertAndGetId
+import org.jetbrains.exposed.sql.selectAll
+import org.jetbrains.exposed.sql.tests.DatabaseTestsBase
+import org.jetbrains.exposed.sql.tests.shared.assertEqualLists
+import org.jetbrains.exposed.sql.tests.shared.assertEquals
+import org.junit.Test
+import kotlin.test.assertNull
+
+class ColumnTransformTest : DatabaseTestsBase() {
+    @Test
+    fun testSimpleTransforms() {
+        val tester = object : IntIdTable("SimpleTransforms") {
+            val stringToInteger = integer("stringToInteger")
+                .transform(toColumn = { it.toInt() }, toReal = { it.toString() })
+            val nullableStringToInteger = integer("nullableStringToInteger")
+                .nullable()
+                .transform(toColumn = { it.toInt() }, toReal = { it.toString() })
+            val stringToIntegerNullable = integer("stringToIntegerNullable")
+                .transform(toColumn = { it.toInt() }, toReal = { it.toString() })
+                .nullable()
+        }
+
+        withTables(tester) {
+            val id1 = tester.insertAndGetId {
+                it[stringToInteger] = "1"
+                it[nullableStringToInteger] = "2"
+                it[stringToIntegerNullable] = "3"
+            }
+            val entry1 = tester.selectAll().where { tester.id eq id1 }.first()
+            assertEquals("1", entry1[tester.stringToInteger])
+            assertEquals("2", entry1[tester.nullableStringToInteger])
+            assertEquals("3", entry1[tester.stringToIntegerNullable])
+
+            val id2 = tester.insertAndGetId {
+                it[stringToInteger] = "1"
+                it[nullableStringToInteger] = null
+                it[stringToIntegerNullable] = null
+            }
+            val entry2 = tester.selectAll().where { tester.id eq id2 }.first()
+            assertEquals(null, entry2[tester.nullableStringToInteger])
+            assertEquals(null, entry2[tester.stringToIntegerNullable])
+        }
+    }
+
+    @Test
+    fun testNestedTransforms() {
+        val tester = object : IntIdTable("NestedTransforms") {
+            val booleanToInteger = integer("stringToInteger")
+                .transform(toReal = { if (it != 0) "TRUE" else "FALSE" }, toColumn = { if (it == "TRUE") 1 else 0 })
+                .transform(toReal = { it == "TRUE" }, toColumn = { if (it) "TRUE" else "FALSE" })
```

Comment:
To confirm, if multiple `transform()`s are chained, the last does not override the previous ones? Instead, the last transformation acts on the transformation before that and so on?

Just wondering if that's the behavior that is wanted, mostly because `EntityClass.transform()` only allows 1 transformation on every lookup of a column's value. But that is less important if the decision is to treat the DSL and DAO column transformations as completely separate.

Reply:
Yes, comparing to DAO's transform, in DSL `transform()` returns column with a new type, so new transform over it should be created with transform functions for the new type.

In order to make it more consistent, I added chained transforms to DAO also. But for DAO it's simpler anyway, because there is no something like column replacement, so I just recreate `ColumnWithTransform` with updated transformation functions. 

And also to allow reusing the same transformers (if the user wants to create a transformer like an object and reuse across the code) DAO's `ColumnWithTransform` accepts `ColumnTransformer` internally.